src/v/kafka/server/consumer_group_lag_metrics_rpc_types.h

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2025 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+#include "container/chunked_hash_map.h"
+#include "model/fundamental.h"
+#include "serde/envelope.h"
+#include "serde/rw/envelope.h"
+#include "serde/rw/map.h"
+#include "serde/rw/set.h"
+
+namespace kafka {
+
+struct partition_offsets_request
+  : serde::envelope<
+      partition_offsets_request,
+      serde::version<0>,
+      serde::compat_version<0>> {
+    using rpc_adl_exempt = std::true_type;
+
+    using partitions
+      = chunked_hash_map<model::topic, chunked_hash_set<model::partition_id>>;
+    partitions data;
+
+    auto serde_fields() { return std::tie(data); }
+};
+
+struct partition_offsets_reply
+  : serde::envelope<
+      partition_offsets_reply,
+      serde::version<0>,
+      serde::compat_version<0>> {
+    using rpc_adl_exempt = std::true_type;
+
+    using offsets = chunked_hash_map<
+      model::topic,
+      chunked_hash_map<model::partition_id, kafka::offset>>;
```

Comment:
should `kafka::offsets` be replaced by a versioned serde type?

Reply:
Maybe I'm not being very creative, but I can't think of anything other than partition hwm that we would need from other nodes when computing consumer lag.

Reply:
> partition hwm that we would need from other nodes when computing consumer lag.

the point of serde is that you don't have to be creative today; you can avoid the pain later of evolving a non-serde rpc.